ADALiOS/ADALiOS/ADAuthenticationContext+TokenCaching.m

```diff
@@ -93,7 +94,7 @@ - (ADTokenCacheStoreItem*)findCacheItemWithKey:(ADTokenCacheStoreKey*) key
         else
         {
             //We have a cache item that cannot be used anymore, remove it from the cache:
-            [self.tokenCacheStore removeItemWithKey:key userId:userId error:nil];
+            [self.tokenCacheStore removeItemWithKey:key userId:userId.userId error:nil];
```

Comment:
That would require a significant rearchitecture of our caching code. Right now we stash the username in the appropriate field in keychain, we would need more keychain properties.